utils.py

```diff
@@ -0,0 +1,294 @@
+from random import choice
+
+import networkx as nx
+from matplotlib import pyplot as plt
+
+from graph.elements.edge import Edge
+from graph.elements.hyperedge import HyperEdge
+from graph.elements.node import Node
+from graph.hypergraph import HyperGraph
+
+
+def equals(graph1: HyperGraph, graph2: HyperGraph) -> bool:
+    return check(graph1, graph2) and check(graph2, graph1)
+
+
+def check(graph: HyperGraph, subgraph: HyperGraph) -> bool | dict:
+    return _find_subgraph(graph, subgraph)
+
+
+def _find_subgraph(graph: HyperGraph, subgraph: HyperGraph) -> bool | dict:
+    found_subgraphs = _find_subgraphs(graph, subgraph)
+    return choice(found_subgraphs) if len(found_subgraphs) > 0 else False
+
+
+def _find_subgraphs(graph: HyperGraph, subgraph: HyperGraph) -> list[dict]:
+    matcher = nx.algorithms.isomorphism.GraphMatcher(
+        graph.parse_hypergraph_to_networkx(),
+        subgraph.parse_hypergraph_to_networkx(),
+        node_match=_node_match
+    )
+
+    return list(matcher.subgraph_isomorphisms_iter())
+
+
+def _node_match(v_self, v_left):
```

Comment:
now I think about it, if it is not the class they probably shouldn't be marked with underscore at all because here is no concept of abstracts and polymorphism